src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -629,73 +632,74 @@ private void SetDefaultProjectName()
 
         private async Task EnsureNuGetAndVsProjectAdapterCacheAsync()
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
-
-            if (!_cacheInitialized && IsSolutionOpen)
+            await _initLock.ExecuteNuGetOperationAsync(async () =>
```

Comment:
ProjectSystemCache has lock, the lock here is useless, during open solution, if pmc or UI try to do init, it will return immediately because SolutionOpenRaised is false, no one will wait on this lock.